README.adoc

```diff
@@ -1,97 +1,114 @@
+if the requested artifact contains `-SRC-` in its version string. For such ones, the `srcdeps-core` library is
+called, to build what is necessary and install it to the Maven local repository. In other words, the dependencies
+are built and installed while they are looked up.
+
+=== How to configure for Maven:
+
+==== 1. `.mvn/extensions.xml`
+
+`srcdeps-maven-local-repository` needs to be declared as a Maven Core Extension in `.mvn/extensions.xml` of the maven project.
 
 [source,xml]
 ----
+<?xml version="1.0" encoding="UTF-8"?>
+<extensions>
+  <extension>
+    <groupId>org.l2x6.maven.srcdeps</groupId>
+    <artifactId>srcdeps-maven-local-repository</artifactId>
+    <version>0.0.12-SNAPSHOT</version>
+  </extension>
+</extensions>
+----
+
+==== 2. `.mvn/srcdeps.yaml`
 
-  [...]
-
-  <dependencies>
-    <dependency>
-      <groupId>org.l2x6.maven.srcdeps.itest</groupId>
-      <artifactId>srcdeps-test-artifact</artifactId>
-      <!-- -SRC- is the marker that makes Srcdeps Plugin to interpret the rest of the version
-           string as a scmVersionType, scmVersion pair. In this case scmVersionType is
-           "revision" and scmVersion is "66ea95d890531f4eaaa5aa04a9b1c69b409dcd0b" - i.e.
-           the sha1 of the git commit we want to build as a dependency.
-           scmVersionType values are given by the SCM (Source Code Management) system in use.
-           For git, scmVersionType values are "revision", "tag" or "branch"
-      -->
-      <version>0.0.1-SRC-revision-66ea95d890531f4eaaa5aa04a9b1c69b409dcd0b</version>
-    </dependency>
-  </dependencies>
-
-  <build>
-    <plugins>
-     [...]
-
-      <plugin>
-        <groupId>org.l2x6.maven.srcdeps</groupId>
-        <artifactId>srcdeps-maven-plugin</artifactId>
-        <version>0.0.1</version>
-        <extensions>true</extensions>
-        <configuration>
-          <repositories>
-            <repository>
-              <!-- a build tree unique id chosen by you -->
-              <id>org.hawkular.cmdgw</id>
-              <selectors>
-                <!-- groupId of the dependencies that come from this repo
-                     The syntax may change in the future,
-                     see https://github.com/l2x6/srcdeps-maven-plugin/issues/10 -->
-                <selector>org.hawkular.cmdgw</selector>
-              </selectors>
-              <!-- The url that is passed to checkout mojo of SCM Plugin,
-                   see https://maven.apache.org/scm/maven-scm-plugin/checkout-mojo.html -->
-              <url>scm:git:https://github.com/hawkular/hawkular-command-gateway.git</url>
-            </repository>
-          </repositories>
-        </configuration>
-      </plugin>
-
-    </plugins>
-  </build>
-
-  [...]
+ Various settings have to be configured in `.mvn/srcdeps.yaml` file. Especially, the mapping from groupId's of
+ dependencies to SCM repositiries where their sources reside.
 
+[source,yaml]
 ----
+configModelVersion: 1.0
```

Comment:
Is this the minimal version, "normal" version, or complete version of this configuration file? On this readme, I would present the reader with the absolute minimum. For instance, I would suppose that `sourcesDirectory` defaults to `~/.m2/srcdeps` already, so, I wouldn't need it there.

In any case, it would be useful to point the reader to a place where the options are documented. For instance, I have no idea what `forwardProperties` mean. I could guess that I could specify a "mvn ... -DmyForwardProperty=bla" and the inner build gets access to it, but that's just a guess :)